subsys/bluetooth/mesh/gen_lvl_srv.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <string.h>
+#include <bluetooth/mesh/gen_lvl_srv.h>
+#include <bluetooth/mesh/gen_dtt_srv.h>
+#include "model_utils.h"
+
+static void encode_status(const struct bt_mesh_lvl_status *status,
+			  struct net_buf_simple *buf)
+{
+	net_buf_simple_add_le16(buf, status->current);
+
+	if (status->remaining_time == 0) {
+		return;
+	}
+
+	net_buf_simple_add_le16(buf, status->target);
+	net_buf_simple_add_u8(buf,
+			      model_transition_encode(status->remaining_time));
+}
+
+static void transition_get(struct bt_mesh_lvl_srv *srv,
+			   struct bt_mesh_model_transition *transition,
+			   struct net_buf_simple *buf)
+{
+	if (buf->len == 2) {
+		model_transition_buf_pull(buf, transition);
+	} else {
+		bt_mesh_dtt_srv_transition_get(srv->model, transition);
+	}
+}
+
+static void rsp_status(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+		       const struct bt_mesh_lvl_status *status)
+{
+	BT_MESH_MODEL_BUF_DEFINE(rsp, BT_MESH_LVL_OP_STATUS,
+				 BT_MESH_LVL_MSG_MAXLEN_STATUS);
+	bt_mesh_model_msg_init(&rsp, BT_MESH_LVL_OP_STATUS);
+	encode_status(status, &rsp);
+
+	bt_mesh_model_send(model, ctx, &rsp, NULL, 0);
+}
+
+static void handle_get(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+		       struct net_buf_simple *buf)
+{
+	if (buf->len != BT_MESH_LVL_MSG_LEN_GET) {
+		return;
+	}
+
+	struct bt_mesh_lvl_srv *srv = model->user_data;
+	struct bt_mesh_lvl_status status = { 0 };
+
+	srv->handlers->get(srv, ctx, &status);
+
+	rsp_status(model, ctx, &status);
+}
+
+static void set(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+		struct net_buf_simple *buf, bool ack)
+{
+	if (buf->len != BT_MESH_LVL_MSG_MINLEN_SET &&
+	    buf->len != BT_MESH_LVL_MSG_MAXLEN_SET) {
+		return;
+	}
+
+	struct bt_mesh_lvl_srv *srv = model->user_data;
+	struct bt_mesh_lvl_status status = { 0 };
+	struct bt_mesh_model_transition transition;
+	struct bt_mesh_lvl_set set;
+
+	set.lvl = net_buf_simple_pull_le16(buf);
+	set.new_transaction = tid_check_and_update(
```

Comment:
new_transaction is bool, tid_check_and_update returns s32_t (should be int). 
Is this a correct assignment? Use boolean operators to make it more clear?